### PR TITLE
Release v1.36.0

### PR DESCRIPTION
**Pull request by tanmay-db**
### New Features and Improvements
* Added `databricks_volumes` as data source  ([#3150](https://github.com/databricks/terraform-provider-databricks/pull/3150)).
* Fixed updating owners for UC resources ([#3189](https://github.com/databricks/terraform-provider-databricks/pull/3189)).
* Validated metastore id for databricks_grant and databricks_grants resources ([#3159](https://github.com/databricks/terraform-provider-databricks/pull/3159)).
* Fixed `databricks_connection` regression when creating without owner ([#3186](https://github.com/databricks/terraform-provider-databricks/pull/3186)).
* Allow using empty strings as job parameters ([#3158](https://github.com/databricks/terraform-provider-databricks/pull/3158)).
* Changed type of value field of `JobsHealthRule` to `int64` ([#3215](https://github.com/databricks/terraform-provider-databricks/pull/3215)).

### Documentation Changes
* Various documentation updates ([#3198](https://github.com/databricks/terraform-provider-databricks/pull/3198)).
* Fixed typo in docs ([#3166](https://github.com/databricks/terraform-provider-databricks/pull/3166)).

### Exporter
* Timestamps are now added to log entries ([#3146](https://github.com/databricks/terraform-provider-databricks/pull/3146)).
* Add retries for `Search`, `ReadContext` and `Import` operations when importing the resource ([#3202](https://github.com/databricks/terraform-provider-databricks/pull/3202)).
* Performance improvements for big workspaces ([#3167](https://github.com/databricks/terraform-provider-databricks/pull/3167)).
* Fix generation of cluster policy resources ([#3185](https://github.com/databricks/terraform-provider-databricks/pull/3185)).
* Skip emitting of clusters that come from more cluster sources ([#3161](https://github.com/databricks/terraform-provider-databricks/pull/3161)).

### Internal Changes
* Migrated cluster schema to use the go-sdk struct ([#3076](https://github.com/databricks/terraform-provider-databricks/pull/3076)).
* Updated actions/setup-go to v5 ([#3154](https://github.com/databricks/terraform-provider-databricks/pull/3154)).
* Changed default branch from `master` to `main` ([#3174](https://github.com/databricks/terraform-provider-databricks/pull/3174)).
* Added .codegen.json configuration ([#3180](https://github.com/databricks/terraform-provider-databricks/pull/3180)).
* Used common.Resource consistently throughout the provider ([#3193](https://github.com/databricks/terraform-provider-databricks/pull/3193)).
* Fixed unit test ([#3201](https://github.com/databricks/terraform-provider-databricks/pull/3201)).
* Added test code for job task order ([#3183](https://github.com/databricks/terraform-provider-databricks/pull/3183)).
* Added unit test for `customizable_schema.go` ([#3192](https://github.com/databricks/terraform-provider-databricks/pull/3192)).
* Extended customizable schema with `AtLeastOneOf`, `ExactlyOneOf`, `RequiredWith` ([#3182](https://github.com/databricks/terraform-provider-databricks/pull/3182)).
* Fixed notebook parameters in acceptance test ([#3205](https://github.com/databricks/terraform-provider-databricks/pull/3205)).
* Introduced Generic Settings Resource ([#2997](https://github.com/databricks/terraform-provider-databricks/pull/2997)).
* Suppress diff should apply to new fields added in the same chained call to CustomizableSchema ([#3200](https://github.com/databricks/terraform-provider-databricks/pull/3200)).

### CODE REVIEW

**Comment by tanmay-db** [comment]
Note: while releasing using deco tool, branch cut was 1.35.1 hence it shows that in the branch. I manually updated the version to 1.36.0

**Comment by tanmay-db** [comment]
Integration test is running...

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,41 @@
 # Version changelog
 
+## 1.36.0
+
+### New Features and Improvements
+* Added `databricks_volumes` as data source  ([#3150](https://github.com/databricks/terraform-provider-databricks/pull/3150)).
+* Introduced Generic Settings Resource ([#2997](https://github.com/databricks/terraform-provider-databricks/pull/2997)).
+* Fixed updating owners for UC resources ([#3189](https://github.com/databricks/terraform-provider-databricks/pull/3189)).
+* Validated metastore id for databricks_grant and databricks_grants resources ([#3159](https://github.com/databricks/terraform-provider-databricks/pull/3159)).
+* Updated ([#3192](https://github.com/databricks/terraform-provider-databricks/pull/3192)).
```
Comment (alexott) [inline CHANGELOG.md:10]: This should be in the "Internal changes" section and also we need to have a full description

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+* Suppress diff should apply to new fields added in the same chained call to CustomizableSchema ([#3200](https://github.com/databricks/terraform-provider-databricks/pull/3200)).
+* Extended customizable schema with `AtLeastOneOf`, `ExactlyOneOf`, `RequiredWith` ([#3182](https://github.com/databricks/terraform-provider-databricks/pull/3182)).
```
Comment (alexott) [inline CHANGELOG.md:12]: This is also internal change

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,41 @@
...
+* Fixed `databricks_connection` regression when creating without owner ([#3186](https://github.com/databricks/terraform-provider-databricks/pull/3186)).
+* Allow using empty strings as job parameters ([#3158](https://github.com/databricks/terraform-provider-databricks/pull/3158)).
+* Fixed notebook parameters in acceptance test ([#3205](https://github.com/databricks/terraform-provider-databricks/pull/3205)).
```
Comment (alexott) [inline CHANGELOG.md:15]: and this

**Review by alexott — CHANGES_REQUESTED** [review]

**Review by alexott — APPROVED** [review]

**Review by mgyucht — APPROVED** [review]

**Pull request merged**